### PR TITLE
Added Aura

### PR DESCRIPTION
**Pull request by harikt**
Hi @dantleech , 

I have added aura/di.

I hope I have added things correctly. If not let me know.

### CODE REVIEW

File: benchmarks/AuraDiBench.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use Aura\Di\ContainerBuilder;
+
+/**
+ * @Groups({"aura-di"}, extend=true)
+ */
+class AuraDiBench extends ContainerBenchCase
+{
+    private $container;
+
+    public function benchGetUnoptimized()
```
Comment (dantleech) [inline benchmarks/AuraDiBench.php:20]: I have skipped this benchmark containers that don't "optimize" (not a very good term, basically containers that don't do any definition caching).

``` php
/**
 * @Skip()
 */
public function benchGetUnoptimized()
{
}
```

File: benchmarks/AuraDiBench.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public function benchGetPrototype()
+    {
+        $this->container->newInstance('PhpBench\Benchmarks\Container\Acme\BicycleFactory');
+    }
+
+    public function benchLifecycle()
+    {
+	$this->init();
```
Comment (dantleech) [inline benchmarks/AuraDiBench.php]: only 3 spaces here :)

File: benchmarks/AuraDiBench.php (hunk 1 above)
Comment (dantleech) [inline benchmarks/AuraDiBench.php:20]: Also note that there is currently a [bug](https://github.com/phpbench/phpbench/pull/385) which I discovered today which makes `@Skip` misbehave incase you have any issues.

File: benchmarks/AuraDiBench.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    private $container;
+
+    public function benchGetUnoptimized()
+    {
+        $this->container->get('bicycle_factory');
+    }
+
+    public function benchGetOptimized()
+    {
+        $this->container->get('bicycle_factory');
```
Comment (harikt) [inline benchmarks/AuraDiBench.php:26]: Isn't the optimized should be skipped in that case ?

File: benchmarks/AuraDiBench.php (hunk 2 above)
Comment (harikt) [inline benchmarks/AuraDiBench.php]: do we really need to call init here ?

File: benchmarks/AuraDiBench.php (hunk 3 above)
Comment (dantleech) [inline benchmarks/AuraDiBench.php:26]: The thinking was that to assume containers are optimized (i.e. AuraDI has no "un-optimized" state, its standard configuration is "optimized"). But we should change the terminology as it is confusing.

But at least for now it should be `Unoptimized` that is skipped.

File: benchmarks/AuraDiBench.php (hunk 2 above)
Comment (dantleech) [inline benchmarks/AuraDiBench.php]: Yeah, this tests the cost of instantiating the container and retreving a service.

**Comment by harikt** [comment]
@dantleech is this actually making any test that the instances created are actually same / different objects ?

**Comment by dantleech** [comment]
@harikt no, we are only benchmarking, not unit testing. adding basic tests to assert that we are benchmarking the correct behavior would be nice, but it would intefere with the measurement (in the case of retrieving a service, it would perhaps double the time reported).

**Comment by dantleech** [comment]
Thanks @harikt  :)

**Comment by harikt** [comment]
You are welcome. Probably you could update the Readme with Aura's data :) .

Thank you for merging.

**Pull request merged**